pkg/apis/sparkoperator.k8s.io/v1beta1/types.go

```diff
@@ -244,6 +244,9 @@ type SparkApplicationSpec struct {
 	// BatchScheduler configures which batch scheduler will be used for scheduling
 	// Optional.
 	BatchScheduler *string `json:"batchScheduler,omitempty"`
+	// BatchSchedulerOptions provides fine-grained control on how to batch scheduling.
+	// Optional.
+	BatchSchedulerOptions map[string]string `json:"batchSchedulerOptions,omitempty"`
```

Comment:
Instead of using a map, I suggest creating a `construct` that have dedicated fields for the configuration properties. For example, something like the following:

```
type BatchSchedulerConfiguration struct {
  Queue string `json:"queue,omitempty"`
}

type SparkApplicationSpec struct {
  ...
  BatchSchedulerConfig *BatchSchedulerConfiguration `json:"batchSchedulerConfig,omitempty"`
  ...
}

``` 

This way, it's much easier to perform schema validation.

Reply:
There would be a bunch of attributes if new batch schedulers coming in.

Reply:
That's fine. It's better to use explicit fields than key-value pairs.